### PR TITLE
chore(theming): migrate theming documentation to storybook

### PR DESCRIPTION
**Pull request by austingreendev**
## Description

This PR migrates our theming documentation to storybook. https://garden.zendesk.com/components/utilities and the other main theming documentation already does a great job of showcasing the theming utilities. This PR just moves over the existing storybook examples.

## Checklist

- [ ] ~:ok_hand: design updates are Garden Designer approved (add the
      designer as a reviewer)~
- [x] :globe_with_meridians: demo is up-to-date (`yarn start`)
- [x] :arrow_left: renders as expected with reversed (RTL) direction
- [x] :metal: renders as expected with Bedrock CSS (`?bedrock`)
- [x] :wheelchair: analyzed via [axe](https://www.deque.com/axe/) and evaluated using VoiceOver
- [ ] ~:guardsman: includes new unit tests~
- [ ] ~:memo: tested in Chrome, Firefox, Safari, Edge, and IE11~

### CODE REVIEW

File: packages/theming/stories/4-DefaultTheme.stories.mdx [hunk 1]
```diff
@@ -1,3 +1,10 @@
+import { Meta, Description, Canvas } from '@storybook/addon-docs/blocks';
+import { DefaultThemeExample } from './examples/DefaultThemeExample';
+
+<Meta title="Components/Theming/Elements/DEFAULT_THEME" />
+
+## DEFAULT_THEME
```
Comment (jzempel) [inline packages/theming/stories/4-DefaultTheme.stories.mdx:6]: I would advocate for deleting this file. The sections on the website are much more comprehensive and will be maintained. This location should not be referenced for similar (or competing) information.

File: packages/theming/stories/3-Palette.stories.mdx [hunk 2]
```diff
@@ -1,3 +1,10 @@
+import { Meta, Description, Canvas } from '@storybook/addon-docs/blocks';
+import { ColorPaletteExample } from './examples/ColorPaletteExample';
+
+<Meta title="Components/Theming/Elements/Palette" />
+
+## Palette
```
Comment (jzempel) [inline packages/theming/stories/3-Palette.stories.mdx:6]: I would advocate for deleting this file. The sections on the website are much more comprehensive and will be maintained. This location should not be referenced for similar (or competing) information. In some cases, the info here is incorrect – a problem that was addressed when content was moved to the website.

**Review by jzempel — CHANGES_REQUESTED** [review]

**Review by hzhu — APPROVED** [review]

File: packages/theming/stories/4-DefaultTheme.stories.mdx (hunk 1 above)
Comment (austingreendev) [reply packages/theming/stories/4-DefaultTheme.stories.mdx:6]: Removed.

File: packages/theming/stories/3-Palette.stories.mdx (hunk 2 above)
Comment (austingreendev) [reply packages/theming/stories/3-Palette.stories.mdx:6]: Removed.

**Review by jzempel — APPROVED** [review]

**Pull request merged**